### ISSUE TITLE
Command line arguments to app when testing

### ISSUE THREAD

**Issue by adamski** [wix/detox#181]
I'd like to invoke a switch in the native part of my app when testing, in this case to mock audio input. I've been looking for a way to pass a command line argument to the app but not found anything - is there a way to do this with Detox? If not I'm happy to create a PR.

**Comment by adamski** [comment]
Looking at the code, my proposal would be to add command line arguments here

https://github.com/adamski/detox/blob/master/detox/src/devices/DeviceLoader.js#L22

e.g. 
```
await fbsimctl.launch(simulatorUdid, bundleId, [{"-test", "true"}])
```

**Comment by LeoNatan** [comment]
We've been discussing this internally for a while. In general, I don't like people having `if(testing) { ... } else { ... }` code in their apps, be it native or JS. Just doesn't look like a good idea in long term code management, in addition to localized problems of enabling test code by mistake.

On the JS front, we are working on a solution that allows you to mock stuff at the "compiler" level - during packaging. I'd love to have a similar solution for native, where at `detox build` time, we would be able to perform injections. This has many challenges, such as Swift support, how to inject, etc. I have not given thought on how to approach a solution yet.

**Comment by rotemmiz** [comment]
This feature will be released with Detox 5.5.0
https://github.com/wix/detox/blob/master/docs/APIRef.DeviceObjectAPI.md#6-additional-launch-arguments
